Assets/MixedRealityToolkit.Providers/WindowsMixedReality/XRSDK/Controllers/WindowsMixedRealityXRSDKArticulatedHand.cs

```diff
@@ -0,0 +1,333 @@
+        private void UpdateHandData(InputDevice inputDevice)
+        {
+            Hand hand;
+            if (inputDevice.TryGetFeatureValue(CommonUsages.handData, out hand))
+            {
+                foreach (HandFinger finger in handFingers)
+                {
+                    if (hand.TryGetFingerBones(finger, fingerBones))
+                    {
+                        for (int i = 0; i < fingerBones.Count; i++)
+                        {
+                            TrackedHandJoint trackedHandJoint = ConvertToTrackedHandJoint(finger, i);
+                            Bone bone = fingerBones[i];
+
+                            Vector3 position = Vector3.zero;
+                            Quaternion rotation = Quaternion.identity;
+
+                            if (bone.TryGetPosition(out position) || bone.TryGetRotation(out rotation))
+                            {
+                                // We want input sources to follow the Playspace, so fold in the playspace transform here to
+                                // put the controller pose into world space.
+                                position = MixedRealityPlayspace.TransformPoint(position);
+                                rotation = MixedRealityPlayspace.Rotation * rotation;
+
+                                unityJointPoses[trackedHandJoint] = new MixedRealityPose(position, rotation);
+                            }
+                        }
+
+                        // Unity doesn't provide a palm joint, so we synthesize one here
+                        unityJointPoses[TrackedHandJoint.Palm] = CurrentControllerPose;
+                    }
+                }
+
+                CoreServices.InputSystem?.RaiseHandJointsUpdated(InputSource, ControllerHandedness, unityJointPoses);
+            }
+
+#if WINDOWS_UWP
+            MixedRealityHandTrackingProfile handTrackingProfile = null;
+            MixedRealityInputSystemProfile inputSystemProfile = CoreServices.InputSystem?.InputSystemProfile;
+            if (inputSystemProfile != null)
+            {
+                handTrackingProfile = inputSystemProfile.HandTrackingProfile;
+            }
+
+            if (handTrackingProfile == null || !handTrackingProfile.EnableHandMeshVisualization)
+            {
+                // if hand mesh visualization is disabled make sure to destroy our hand mesh observer if it has already been created
+                if (handMeshObserver != null)
+                {
+                    // Notify that hand mesh has been updated (cleared)
+                    HandMeshInfo handMeshInfo = new HandMeshInfo();
+                    CoreServices.InputSystem?.RaiseHandMeshUpdated(InputSource, ControllerHandedness, handMeshInfo);
+                    hasRequestedHandMeshObserver = false;
+                    handMeshObserver = null;
+                }
+                return;
+            }
+
+            List<object> states = new List<object>();
```

Comment:
>states = new List<object>(); [](start = 25, length = 28)

Is it possible to cache this list object so that we don't do an alloc of this on each frame?

Reply:
I feel like there was some reason I didn't do this initially...i'll take another look at it

Reply:
Moved it. Looked around a bit and caching this is all good